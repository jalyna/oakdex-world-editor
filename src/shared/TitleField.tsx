import * as React from 'react'
import styled from 'styled-components'

import TextField from 'shared/TextField'
import Button from 'shared/Button'
import { DEFAULT_FONT, GREY_30 } from 'shared/theme'

const Wrapper = styled.div`
  font-family: ${DEFAULT_FONT};
  color: ${GREY_30};
  font-size: 18px;
  cursor: pointer;
`

const TextFieldWrapper = styled.div`
  float: left;
  width: 80%;
  padding-right: 20px;
  box-sizing: border-box;
`

const ButtonWrapper = styled.div`
  float: left;
  width: 20%;
`

interface TitleFieldProps {
  title: string,
  onChange: (newTitle: string) => void
}

interface TitleFieldState {
  isEditing: boolean
}

class TitleField extends React.Component<TitleFieldProps, TitleFieldState> {
  constructor (props: TitleFieldProps) {
    super(props)
    this.state = {
      isEditing: false
    }
    this.onChange = this.onChange.bind(this)
    this.onClick = this.onClick.bind(this)
  }

  render () {
    if (this.state.isEditing) {
      return (
        <div>
          <TextFieldWrapper>
            <TextField
              value={this.props.title}
              onChange={this.onChange} />
          </TextFieldWrapper>
          <ButtonWrapper>
            <Button onClick={this.onClick}>
              OK
            </Button>
          </ButtonWrapper>
        </div>
      )
    }

    return (
      <Wrapper onClick={this.onClick}>
        {this.props.title}
      </Wrapper>
    )
  }

  onChange (e: React.FormEvent<HTMLInputElement>) {
    this.props.onChange((e.target as HTMLInputElement).value)
  }

  onClick (e: React.MouseEvent) {
    e.preventDefault()
    this.setState({ isEditing: !this.state.isEditing })
  }
}

export default TitleField
