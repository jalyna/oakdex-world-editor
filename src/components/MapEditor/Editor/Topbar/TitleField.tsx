import * as React from 'react'
import styled from 'styled-components'
import { connect } from 'react-redux'
import { Dispatch } from 'redux'

import { MapData } from 'components/MapEditor/reducers/mapData'
import TextField from 'shared/TextField'
import Button from 'shared/Button'

import { DEFAULT_FONT, GREY_30 } from 'shared/theme'
import { UPDATE_MAP } from 'components/MapEditor/actionTypes'

const Wrapper = styled.div`
  font-family: ${DEFAULT_FONT};
  color: ${GREY_30};
  font-size: 18px;
  cursor: pointer;
`

const TextFieldWrapper = styled.div`
  float: left;
  width: 80%;
  padding-right: 20px;
  box-sizing: border-box;
`

const ButtonWrapper = styled.div`
  float: left;
  width: 20%;
`

interface TitleFieldProps {
  mapData: MapData,
  onChange: (e: React.FormEvent<HTMLInputElement>) => void
}

interface TitleFieldState {
  isEditing: boolean
}

function mapStateToProps ({ mapData }: any) {
  return {
    mapData
  }
}

function mapDispatchToProps (dispatch: Dispatch) {
  return {
    onChange: (e: React.FormEvent<HTMLInputElement>) => {
      dispatch({
        type: UPDATE_MAP,
        data: {
          title: e.currentTarget.value
        }
      })
    }
  }
}

class TitleField extends React.Component<TitleFieldProps, TitleFieldState> {
  constructor (props: TitleFieldProps) {
    super(props)
    this.state = {
      isEditing: false
    }
    this.onChange = this.onChange.bind(this)
    this.onClick = this.onClick.bind(this)
  }

  render () {
    if (this.state.isEditing) {
      return (
        <div>
          <TextFieldWrapper>
            <TextField
              value={this.props.mapData.title}
              onChange={this.onChange} />
          </TextFieldWrapper>
          <ButtonWrapper>
            <Button onClick={this.onClick}>
              OK
            </Button>
          </ButtonWrapper>
        </div>
      )
    }

    return (
      <Wrapper onClick={this.onClick}>
        {this.props.mapData.title}
      </Wrapper>
    )
  }

  onChange (e: React.FormEvent<HTMLInputElement>) {
    this.props.onChange(e)
  }

  onClick (e: React.MouseEvent) {
    e.preventDefault()
    this.setState({ isEditing: !this.state.isEditing })
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(TitleField)
