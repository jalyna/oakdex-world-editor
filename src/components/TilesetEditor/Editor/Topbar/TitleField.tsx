import * as React from 'react'
import styled from 'styled-components'
import { connect } from 'react-redux'
import { Dispatch } from 'redux'

import { Tileset } from 'components/TilesetEditor/reducers/tilesetData'
import TextField from 'shared/TextField'
import Button from 'shared/Button'

import { DEFAULT_FONT, GREY_30 } from 'shared/theme'
import { UPDATE_TILESET } from 'components/TilesetEditor/actionTypes'

const Wrapper = styled.div`
  font-family: ${DEFAULT_FONT};
  color: ${GREY_30};
  font-size: 18px;
  cursor: pointer;
`

const TextFieldWrapper = styled.div`
  float: left;
  width: 80%;
  padding-right: 20px;
  box-sizing: border-box;
`

const ButtonWrapper = styled.div`
  float: left;
  width: 20%;
`

interface TitleFieldProps {
  tilesetData: Tileset,
  onChange: (e: React.FormEvent<HTMLInputElement>) => void
}

interface TitleFieldState {
  isEditing: boolean
}

function mapStateToProps ({ tilesetData }: any) {
  return {
    tilesetData
  }
}

function mapDispatchToProps (dispatch: Dispatch) {
  return {
    onChange: (e: React.FormEvent<HTMLInputElement>) => {
      dispatch({
        type: UPDATE_TILESET,
        data: {
          title: e.currentTarget.value
        }
      })
    }
  }
}

class TitleField extends React.Component<TitleFieldProps, TitleFieldState> {
  constructor (props: TitleFieldProps) {
    super(props)
    this.state = {
      isEditing: false
    }
    this.onChange = this.onChange.bind(this)
    this.onClick = this.onClick.bind(this)
  }

  render () {
    if (this.state.isEditing) {
      return (
        <div>
          <TextFieldWrapper>
            <TextField
              value={this.props.tilesetData.title}
              onChange={this.onChange} />
          </TextFieldWrapper>
          <ButtonWrapper>
            <Button onClick={this.onClick}>
              OK
            </Button>
          </ButtonWrapper>
        </div>
      )
    }

    return (
      <Wrapper onClick={this.onClick}>
        {this.props.tilesetData.title}
      </Wrapper>
    )
  }

  onChange (e: React.FormEvent<HTMLInputElement>) {
    this.props.onChange(e)
  }

  onClick (e: React.MouseEvent) {
    e.preventDefault()
    this.setState({ isEditing: !this.state.isEditing })
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(TitleField)
