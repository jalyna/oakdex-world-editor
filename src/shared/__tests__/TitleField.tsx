import * as React from 'react'
import { mount } from 'enzyme'
import TitleField from '../TitleField'

interface WithValue {
  value: string
}

describe('<TitleField>', () => {
  const onChange = jest.fn()

  it('renders correctly', () => {
    const wrapper = mount(
      <TitleField title='foobar' onChange={onChange} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  describe('when clicking on title', () => {
    it('shows text input', () => {
      const wrapper = mount(
        <TitleField title='foobar' onChange={onChange} />
      )
      expect(wrapper.find('TextField')).toHaveLength(0)
      wrapper.simulate('click')
      expect(wrapper.find('TextField').prop('value')).toEqual('foobar')
      wrapper.find('TextField').simulate('change', { target: { value: 'New Title' } })
      expect(onChange).toHaveBeenCalledWith('New Title')
      wrapper.find('Button').simulate('click')
      expect(wrapper.find('TextField')).toHaveLength(0)
    })
  })
})
