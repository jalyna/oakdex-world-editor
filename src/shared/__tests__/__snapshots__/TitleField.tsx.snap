// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleField> renders correctly 1`] = `
<TitleField
  onChange={[MockFunction]}
  title="foobar"
>
  <styled.div
    onClick={[Function]}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "lastClassName": "jIrAmg",
            "rules": Array [
              "
  font-family: ",
              "'Roboto', sans-serif",
              ";
  color: ",
              "#4d4d4d",
              ";
  font-size: 18px;
  cursor: pointer;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
    >
      <div
        className="sc-htpNat jIrAmg"
        onClick={[Function]}
      >
        foobar
      </div>
    </StyledComponent>
  </styled.div>
</TitleField>
`;
